tooling/nargo/src/ops/test.rs

```diff
@@ -278,85 +266,41 @@ fn check_expected_failure_message(
+        let executor = foreign_call_executor(output, layers::Unhandled);
+
+        Self { executor, encountered_unknown_foreign_call: false }
     }
 }
 
-impl<'a, F: AcirField + Serialize + for<'b> Deserialize<'b>> ForeignCallExecutor<F>
-    for TestForeignCallExecutor<'a, F>
+impl<E, F> ForeignCallExecutor<F> for TestForeignCallExecutor<E>
+where
+    F: AcirField,
+    E: ForeignCallExecutor<F>,
 {
     fn execute(
         &mut self,
         foreign_call: &ForeignCallWaitInfo<F>,
     ) -> Result<ForeignCallResult<F>, ForeignCallError> {
         // If the circuit has reached a new foreign call opcode then it can't have failed from any previous unknown foreign calls.
         self.encountered_unknown_foreign_call = false;
-
-        let foreign_call_name = foreign_call.function.as_str();
-        match ForeignCall::lookup(foreign_call_name) {
-            Some(ForeignCall::Print) => self.printer.execute(foreign_call),
-
-            Some(
-                ForeignCall::CreateMock
-                | ForeignCall::SetMockParams
-                | ForeignCall::GetMockLastParams
-                | ForeignCall::SetMockReturns
-                | ForeignCall::SetMockTimes
-                | ForeignCall::ClearMock,
-            ) => self.mocker.execute(foreign_call),
-
-            None => {
-                // First check if there's any defined mock responses for this foreign call.
-                match self.mocker.execute(foreign_call) {
-                    Err(ForeignCallError::NoHandler(_)) => (),
-                    response_or_error => return response_or_error,
-                };
-
-                if let Some(external_resolver) = &mut self.external {
-                    // If the user has registered an external resolver then we forward any remaining oracle calls there.
-                    match external_resolver.execute(foreign_call) {
-                        Err(ForeignCallError::NoHandler(_)) => (),
-                        response_or_error => return response_or_error,
-                    };
-                }
-
+        match self.executor.execute(foreign_call) {
+            Err(ForeignCallError::NoHandler(_)) => {
                 self.encountered_unknown_foreign_call = true;
-
-                // If all executors have no handler for the given foreign call then we cannot
-                // return a correct response to the ACVM. The best we can do is to return an empty response,
-                // this allows us to ignore any foreign calls which exist solely to pass information from inside
-                // the circuit to the environment (e.g. custom logging) as the execution will still be able to progress.
-                //
-                // We optimistically return an empty response for all oracle calls as the ACVM will error
-                // should a response have been required.
-                Ok(ForeignCallResult::default())
+                layers::Empty.execute(foreign_call)
```

Comment:
```suggestion
 // If all executors have no handler for the given foreign call then we cannot
                // return a correct response to the ACVM. The best we can do is to return an empty response,
                // this allows us to ignore any foreign calls which exist solely to pass information from inside
                // the circuit to the environment (e.g. custom logging) as the execution will still be able to progress.
                //
                // We optimistically return an empty response for all oracle calls as the ACVM will error
                // should a response have been required.
                layers::Empty.execute(foreign_call)
```

Reply:
Maybe something shorter as this paragraph is now in documentation for `Empty` but we want to explain why the `TestForeignCallExecutor` is doing what it does.